### PR TITLE
Add "testseq.py" to Scripts

### PR DESCRIPTION
**Pull request by Adil-Iqbal**
Add "testseq" function.

### CODE REVIEW

**Comment by peterjc** [comment]
Rather than assuming ``/`` as a path separator for splitting (which will break on Windows), use ``os.path.split(...)`` and ``os.path.join(...)`` for the path manipulations. Or just assume that it will be run from the ``Tests/`` folder and hard-code the relative path ``../Scripts/`` (which should work on Windows which will accept the other slashes as input)?

**Comment by andrewguy** [comment]
I'm still concerned that the random number generation is a little unconventional, and more confusing than it needs to be.

I understand your concern that the code needs to be intuitive for those unfamiliar with coding in general, but an equally valid concern here is code readability for anyone who reads the source code. Random number generation should be straightforward, and there are fairly well established ways of doing it.

As written, you're seeding a random number generator with the output from _another_ random number generator, every time you make a call to `random()`.

Can we maybe get another set of eyes on this? This is what I imagine it should look like, but happy for people to disagree - https://github.com/Adil-Iqbal/Personal-Projects/pull/1/files

**Comment by Adil-Iqbal** [comment]
That's a very good point. 

No sense trying to reinvent the wheel. Besides, if this code is to be used in an educational capacity, no sense trying to reinforce concepts that would have to be unlearned.   

I'll see about reverting the seeding method.

**Comment by Adil-Iqbal** [comment]
The following was copy/pasted from the mailing list:

**TL:DR - Game plan: (1) Resolve unittest importing issue. (2) Revert testseq seeding. (3) Revisit testseq doctests. (4) Update unittest for reverted seeding. (2) Thank you for your patience.**

Hello all, 

I'd like to let everyone know where 'testseq' is in development. You can find the code for it here: https://github.com/Adil-Iqbal/Personal-Projects/tree/master/Test%20Sequence

The biggest issue (unexpectedly) was importing the 'testseq' module into the 'test_testseq' unit test. Absolute and relative importing did not work since the 'Scripts' folder is not a part of the 'Bio' package. After trying out several solutions, the one that looks the most promising is using the "importlib" module. The issue is not yet resolved but I am confident that it is just a matter of working out the details.

I've gotten the chance to tackle part of another big issue, which is python version and operating system compatibility. Since the 'testseq' function uses randomization, its output can be different across versions and platforms. This poses a problem for code testing. Luckily there are a few work-arounds that I've implemented in the unit test which should work fine. I will also be revisiting the 'testseq' doctests to ensure they are also compatible, though I have not yet run into any issues with that on Travis CI. (Peter warned me about this. ^_^)

Today, an argument was presented for reverting the seeding implementation, which was that user intuition should preferably be developed with respect to best practices. Unless anyone has a reasonable rebuttal, I will probably revert the seeding to an earlier state. Though, that will have to wait until after I've resolved the importing issue.

Also, I'd like to thank you for being patient with me. I would've completed this sooner, but I have other obligations as well. 

Warmest regards,
Adil

File: Tests/run_tests.py [hunk 1]
```diff
@@ -173,7 +170,6 @@ def is_numpy():
 # Skip Bio.bgzf doctest for broken gzip, see http://bugs.python.org/issue17666
 def _have_bug17666():
     """Debug function to check if Python's gzip is broken (PRIVATE).
-
```
Comment (peterjc) [inline Tests/run_tests.py]: Please restore this blank line which is a deliberate part of the docstring for private function ``_have_bug17666`` - see https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings

**Comment by peterjc** [comment]
Travis is failing on the style checks before ever running the tests:

```
$ flake8 Tests/
Tests/run_tests.py:1:1: D205 1 blank line required between summary line and description
Tests/run_tests.py:171:1: D205 1 blank line required between summary line and description
Tests/run_tests.py:323:1: D205 1 blank line required between summary line and description
```

You can install flake8 locally, see https://github.com/biopython/biopython/blob/master/CONTRIBUTING.rst

**Comment by andrewguy** [comment]
Your latest build is failing with Python 2.7 because you are doing division, and haven't added `from __future__ import division`. Python 2.7 division defaults to floor division, whereas Python 3+ is floating point division.

I would really recommend testing what you can on your local machine before pushing changes to github.

You can use either pip's virtualenv, or conda's env to set up self-contained distributions for testing (if you haven't already done so). You can set up a Python 2.7 environment this way, and make sure your build passes on both 2.7 and 3+. 

Peter's suggestion of running Flake 8 locally is also a very good one.

**Comment by Adil-Iqbal** [comment]
Thanks a lot guys. I really appreciate it. I've got flake8 up and running locally. I'll try to get virtualenv going too.

File: Tests/run_tests.py [hunk 2]
```diff
@@ -7,9 +7,7 @@
 This will find all modules whose name is "test_*.py" in the test
 directory, and run them.  Various command line options provide
 additional facilities.
-
```
Comment (peterjc) [inline Tests/run_tests.py]: You have accidentally remove this (and two more) blank line from the module's docstring.

**Comment by peterjc** [comment]
I wonder if the complicated import tricks you are trying in order to be able to do ``from Scripts import testseq`` or similar could be avoided by including ``Scripts/`` in ``sys.path`` in the same way that ``run_tests.py`` ensures our complied C modules can be imported from the ``build/`` subdirectory?

File: Tests/run_tests.py (hunk 2 above)
Comment (Adil-Iqbal) [inline Tests/run_tests.py]: Reverted. Will include changes in next commit.

**Comment by Adil-Iqbal** [comment]
That method will certainly succeed in importing the module. The issue is that once the importing is complete, the doctests are run. And the doctest *also* contain an import statement. That import statement is subject to the same importing limitations which I will describe below:

Prior to 3.3, heirarchical importing (importing with "dot" notation) required a directory to be initialized with a "__init__.py" file. ([Source](https://docs.python.org/2/tutorial/modules.html#packages)) This created a bunch of confusion, so after 3.3, heirarchical importing was generalized to all directories, regardless of the directory being initialized. 

Since we can't add a "__init__.py" file to folders carelessly, its best to remove non-Bio modules from the "run_tests.py" file prior to version 3.3.  That's why the most recent build passed the Travis CI test. Thankfully, the unit test for 'testseq' runs all of the same tests that are in the doctests, so we can be confident that 'testseq' is compatible with all versions of python. 

All versions of python after 3.3 don't require an extra "__init__.py" file, so the doctests will run fine on 3.3, 3.4, 3.5, 3.6, and PyPy3.

**Comment by peterjc** [comment]
I'm pretty sure you can remove these lines from ``testseq.py`` and the doctest will still work as the classes etc are defined in the current file so already present in the local scope:

```python
>>> from Scripts.testseq import testseq
```

**Comment by Adil-Iqbal** [comment]
Closing pull request.
